sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateSchemaCompatibilityChecker.scala

```diff
@@ -176,11 +179,18 @@ class StateSchemaCompatibilityChecker(
       true
     } else {
       // validate if the new schema is compatible with the existing schema
-      existingStateSchemaList.lazyZip(newStateSchemaList).foreach {
-        case (existingStateSchema, newStateSchema) =>
-          check(existingStateSchema, newStateSchema, ignoreValueSchema)
+      val existingSchemaMap = existingStateSchemaList.map { schema =>
```

Comment:
How is this part of the change related ?

Reply:
It's because the deleteIfExists test case will fail because lazyZip compares the schema by position in the list, so what ends up happening is [countState, mostRecent] gets compared with [mostRecent] after countState is deleted, and this is returned as invalid.

Reply:
Could we add this comment in the code please ?